snuba/utils/registered_class.py

```diff
@@ -0,0 +1,55 @@
+from abc import ABCMeta
+from typing import Any, Dict, Generic, Optional, Tuple, Type, TypeVar, cast
+
+T = TypeVar("T")
+
+
+class _ClassRegistry(Generic[T]):
+    """Keep a mapping of classes to their names"""
+
+    def __init__(self) -> None:
+        self.__mapping: Dict[str, RegisteredClass] = {}
+
+    def register_class(self, cls: "RegisteredClass") -> None:
+        key = cls.config_key()
+        existing_class = self.__mapping.get(key)
+        if not existing_class:
+            self.__mapping[key] = cls
+
+    def get_class_from_name(self, cls_name: str) -> Optional["RegisteredClass"]:
```

Comment:
It could be handled by the caller but in general I think this should be a hard failure.

Reply:
fair enough, I'll make that change